File: leetcode_problems/typescript/src/lc374_guess_number_higher_or_lower.ts
function guess(num: number) {
  return 0;
}

function guessNumber(n: number): number {
  let low = 0;
  let high = n;
  while (high - low > 1) {
    let mid = Math.floor(low + (high - low) / 2);
    let guessed = guess(mid);
    if (guessed == 0) {
      return mid;
    } else if (guessed == 1) {
      low = mid;
    } // means -1 => higher
    else {
      high = mid;
    }
  }
  return high;
}

File: leetcode_problems/cpp/lc374_guess_number_higher_or_lower.cpp
int guess(int num)
{
    return 0;
}

class Solution
{
public:
    int guessNumber(int n)
    {
        int low = 0;
        int high = n;
        while (high - low > 1)
        {
            int mid = low + (high - low) / 2;
            int guessed = guess(mid);
            if (guessed == 0)
            {
                return mid;
            }
            else if (guessed == 1)
            {
                low = mid;
            }
            else // means -1 => higher
            {
                high = mid;
            }
        }
        return high;
    }
};